src/linear_algebra/contraction.lean

```diff
@@ -0,0 +1,61 @@
+/-
+Copyright (c) 2020 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import linear_algebra.tensor_product
+import linear_algebra.dual
+
+/-!
+# Contractions
+
+Given modules $M, N$ over a commutative ring $R$, this file defines the natural linear maps:
+$M^* \otimes M \to R$, $M \otimes M^* \to R$, and $M^* \otimes N → Hom(M, N)$, as well as proving
+some basic properties of these maps.
+
+## Tags
+
+contraction, dual module, tensor product
+-/
+
+universes u v
+
+set_option class.instance_max_depth 42
+
+namespace contraction
```

Comment:
I'm not sure if this is the right name for this namespace. Maybe just make it a section for now? Of put it in `linear_map`?

Reply:
On reflection I agree: not worth a new namespace just for this, and probably best just as a section, which is what I have now done.

Looking at the contenders, `linear_map` probably would be the best fit:
```
~ cat src/linear_algebra/*lean | grep '^namespace' | sort | uniq

namespace alt_bilin_form
namespace alt_sesq_form
namespace bilin_form
namespace direct_sum
namespace dual
namespace dual_pair
namespace equiv
namespace finite_dimensional
namespace finsupp
namespace general_linear_group
namespace is_basis
namespace is_linear_map
namespace linear_action
namespace linear_equiv
namespace linear_map
namespace matrix
namespace module
namespace multilinear_map
namespace pi
namespace quotient
namespace refl_bilin_form
namespace refl_sesq_form
namespace relators
namespace sesq_form
namespace submodule
namespace sym_bilin_form
namespace sym_sesq_form
namespace tensor_product
namespace vector_space
```